net/fetcher.go

```diff
@@ -15,7 +15,9 @@ import (
 
 // Fetcher defines an interface that may be used to fetch data from the network.
 type Fetcher interface {
-	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, recur int) ([]types.TipSet, error)
+	// FetchTipSets will only fetch TipSets whos TipSetKeys evaluate to `false` when passed to `done`,
+	// this includes the provided `tsKey`. The returns slice of TipSets is in Traversal order.
+	FetchTipSets(ctx context.Context, tsKey types.TipSetKey, done func(ts types.TipSet) (bool, error)) ([]types.TipSet, error)
```

Comment:
The `done` callback now returns an error. This is because we are performing a finality check with it and that method can error.

Reply:
Blocking: this should always return at least one tipset, even if the predicate is `true`. The predicate means done *after* this tipset.

Reply:
agreed, addressed.